drivers/serial/uart_native_tty.c

```diff
@@ -157,6 +175,148 @@ static int native_tty_configure(const struct device *dev, const struct uart_conf
 	return native_tty_configure_bottom(fd, &bottom_cfg);
 }
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static int native_tty_uart_fifo_fill(const struct device *dev,
+				     const uint8_t *tx_data,
+				     int size)
+{
+	struct native_tty_data *data = dev->data;
+
+	return nsi_host_write(data->fd, (void *)tx_data, size);
+}
+
+static int native_tty_uart_fifo_read(const struct device *dev,
+				     uint8_t *rx_data,
+				     const int size)
+{
+	struct native_tty_data *data = dev->data;
+
+	return nsi_host_read(data->fd, rx_data, size);
+}
+
+static int native_tty_uart_irq_tx_ready(const struct device *dev)
+{
+	struct native_tty_data *data = dev->data;
+
+	return data->tx_irq_enabled ? 1 : 0;
+}
+
+static int native_tty_uart_irq_tx_complete(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	return 1;
+}
+
+static void native_tty_uart_irq_tx_enable(const struct device *dev)
+{
+	struct native_tty_data *data = dev->data;
+
+	data->tx_irq_enabled = true;
+}
+
+static void native_tty_uart_irq_tx_disable(const struct device *dev)
+{
+	struct native_tty_data *data = dev->data;
+
+	data->tx_irq_enabled = false;
+}
+
+static void native_tty_uart_irq_rx_enable(const struct device *dev)
+{
+	struct native_tty_data *data = dev->data;
+
+	data->rx_irq_enabled = true;
+}
+
+static void native_tty_uart_irq_rx_disable(const struct device *dev)
+{
+	struct native_tty_data *data = dev->data;
+
+	data->rx_irq_enabled = false;
+}
+
+static int native_tty_uart_irq_rx_ready(const struct device *dev)
+{
+	struct native_tty_data *data = dev->data;
+
+	if (data->rx_irq_enabled && native_tty_poll_bottom(data->fd) == 1) {
+		return 1;
+	}
+	return 0;
+}
+
+static int native_tty_uart_irq_is_pending(const struct device *dev)
+{
+	return native_tty_uart_irq_rx_ready(dev) ||
+		native_tty_uart_irq_tx_ready(dev);
+}
+
+static int native_tty_uart_irq_update(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	return 1;
+}
+
+static void native_tty_uart_irq_handler(const struct device *dev)
+{
+	struct native_tty_data *data = dev->data;
+
+	if (data->callback) {
+		data->callback(dev, data->cb_data);
+	} else {
+		WARN("No callback!\n");
+	}
+}
+
+/*
+ * Emulate uart interrupts using a polling thread
+ */
+void native_tty_uart_irq_function(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+	struct device *dev = (struct device *)arg1;
+	struct native_tty_data *data = dev->data;
+
+	while (1) {
+		if (data->rx_irq_enabled) {
+			int ret = native_tty_poll_bottom(data->fd);
+
+			if (ret == 1) {
+				native_tty_uart_irq_handler(dev);
+			} else if (ret < 0) {
+				WARN("Poll returned error %d\n", ret);
+			} else {
+				k_sleep(K_MSEC(1));
+			}
+		} else if (data->tx_irq_enabled) {
```

Comment:
Is it possible that this should be an 'if' instead of an 'else if'? rx_irq_enabled is only set false while reading the receive buffer, so data could never be transmitted in my case.

Reply:
It seems like a bug. @ct-fk Would you like to send a PR with the fix?